includes/class-amp-theme-support.php

```diff
@@ -310,65 +319,174 @@ public static function purge_amp_query_vars() {
 		}
 	}
 
+	/**
+	 * Send an HTTP response header.
+	 *
+	 * This largely exists to facilitate unit testing but it also provides a better interface for sending headers.
+	 *
+	 * @since 0.7.0
+	 *
+	 * @param string $name  Header name.
+	 * @param string $value Header value.
+	 * @param array  $args {
+	 *     Args to header().
+	 *
+	 *     @type bool $replace     Whether to replace a header previously sent. Default true.
+	 *     @type int  $status_code Status code to send with the sent header.
+	 * }
+	 * @return bool Whether the header was sent.
+	 */
+	public static function send_header( $name, $value, $args = array() ) {
+		$args = array_merge(
+			array(
+				'replace'     => true,
+				'status_code' => null,
+			),
+			$args
+		);
+
+		self::$headers_sent[] = array_merge( compact( 'name', 'value' ), $args );
+		if ( headers_sent() ) {
+			return false;
+		}
+
+		header(
+			sprintf( '%s: %s', $name, $value ),
+			$args['replace'],
+			$args['status_code']
+		);
+		return true;
+	}
+
 	/**
 	 * Hook into a form submissions, such as the comment form or some other form submission.
 	 *
 	 * @since 0.7.0
-	 * @global string $pagenow
 	 */
 	public static function handle_xhr_request() {
-		global $pagenow;
-		if ( empty( self::$purged_amp_query_vars['__amp_source_origin'] ) ) {
+		if ( empty( self::$purged_amp_query_vars['__amp_source_origin'] ) || empty( $_SERVER['REQUEST_METHOD'] ) || 'POST' !== $_SERVER['REQUEST_METHOD'] ) {
 			return;
 		}
 
-		if ( isset( $pagenow ) && 'wp-comments-post.php' === $pagenow ) {
-			// We don't need any data, so just send a success.
-			add_filter( 'comment_post_redirect', function() {
-				// We don't need any data, so just send a success.
-				wp_send_json_success();
-			}, PHP_INT_MAX );
-			self::handle_xhr_headers_output();
-		} elseif ( ! empty( self::$purged_amp_query_vars['_wp_amp_action_xhr_converted'] ) ) {
```

Comment:
I realized that we're now not properly looking at `_wp_amp_action_xhr_converted`.

Reply:
On further thought, I don't think the logic this function cares if `_wp_amp_action_xhr_converted`. If someone is implementing their own `action-xhr` endpoint, then they're not going to be doing `wp_redirect()` anyway, but rather should be responding with their own appropriate `wp_send_json()` call.